src/pybind/mgr/dashboard/frontend/src/app/shared/components/delete-confirmation-modal/delete-confirmation-modal.component.ts

```diff
@@ -63,13 +66,27 @@ export class DeleteConfirmationModalComponent extends BaseModal implements OnIni
         ])
       ])
     };
+
     if (this.childFormGroup) {
       controls['child'] = this.childFormGroup;
     }
     this.deletionForm = new CdFormGroup(controls);
     if (!(this.submitAction || this.submitActionObservable)) {
       throw new Error('No submit action defined');
     }
+    if (this.bodyContext?.disable) {
+      this.disableFormControls();
+      return;
+    }
+
+    if (this.bodyContext?.requireExactMatch && this.itemNames?.[0]) {
+      const target = String(this.itemNames[0]);
+      const updateDisabled = (val: string) => {
+        this.submitDisabled = val !== target;
+      };
+      updateDisabled(this.deletionForm.controls.confirmInput.value);
+      this.deletionForm.controls.confirmInput.valueChanges.subscribe((val) => updateDisabled(val));
```

Comment:
this subscribe can lead to memory leak if its not being properly closed btw and I don't see it getting closed here..

maybe we can use the rxjs better to improve this? if you convert the `submitDisabled` boolean to an observable like `submitDisabled$`, then

```
const target = String(this.itemNames[0]);
const confirmControl = this.deletionForm.controls.confirmInput;

this.submitDisabled$ = confirmControl.valueChanges.pipe(
    startWith(confirmControl.value),
    map((value: string) => value !== target)
);
```

and then in html, you just need to give
```
[disabled="submitDisabled$ | async || deletionForm.disabled"
```

and that can do the trick probably?

Also move it to its own function rather.